src/ompl/geometric/planners/rrt/RRTstar.h

```diff
@@ -148,42 +160,126 @@ namespace ompl
+            bool getSampleRejection() const
+            {
+                return useRejectionSampling_;
+            }
+
+            /** \brief Controls where heuristic rejection is used on new states before connection (e.g., x_new = steer(x_nearest, x_rand)) */
+            void setNewStateRejection(const bool reject)
+            {
+                useNewStateRejection_ = reject;
+            }
+
+            /** \brief Get the state of the new-state rejection option */
+            bool getNewStateRejection() const
+            {
+                return useNewStateRejection_;
+            }
+
+            /** \brief Controls search focusing. Search focusing consists of pruning the existing search and limiting future search.
+            In RRT* this is accomplished by turning on Informed RRT* (tree pruning and focused sampling) and new-state rejection.
+            This flag individually sets the options described above.
+            */
+            void setFocusSearch(const bool focus)
+            {
+                setInformedRrtStar(focus);
+                setNewStateRejection(focus);
+            }
+
+            /** \brief Get the state of search focusing */
+            bool getFocusSearch() const
+            {
+                return getTreePruning() && getSampleRejection() && getNewStateRejection();
+            }
+
+            /** \brief Use a k-nearest search for rewiring instead of a r-disc search. */
+            void setKNearest(bool useKNearest)
+            {
+                useKNearest_ = useKNearest;
+            }
+
+            /** \brief Get the state of using a k-nearest search for rewiring. */
+            bool getKNearest() const
+            {
+                return useKNearest_;
+            }
+
+            /** \brief Use \e Informed \e RRT*.
+            This means that once a problem is found, the search is focused only to the subproblem that could contain a better solution.
+            Currently only implemented for problems with a single goal that are seeking to minimize path length in
+            R^n (i.e., RealVectorStateSpace), SE(2) (i.e., SE2StateSpace), or SE(3) (i.e., SE3StateSpace).
+            @par J D. Gammell, S. S. Srinivasa, T. D. Barfoot, "Informed RRT*: Optimal Sampling-based
+            Path Planning Focused via Direct Sampling of an Admissible Ellipsoidal Heuristic."
+            IROS 2014. DOI: <a href="http://dx.doi.org/10.1109/IROS.2014.6942976">10.1109/IROS.2014.6942976</a>.
+            <a href="http://www.youtube.com/watch?v=d7dX5MvDYTc">Illustration video</a>.
+            <a href="http://www.youtube.com/watch?v=nsl-5MZfwu4">Short description video</a>. */
+            void setInformedRrtStar(bool informedRrtStar);
+
+            /** \brief Get the state of informed RRT* */
+            bool getInformedRrtStar() const
+            {
+                return useInformedSampling_ && getTreePruning();
+            }
```

Comment:
I think it is easier for users if they can construct a planner type Foo, rather than RRTstar and then set a bunch of things to make it actually be InformedRRTStar. How about we add an InformedRRTStar, which is like PRMstar -- essentially just a new name + one or two settings in the base class (so all the code remains in RRTstar)